### PR TITLE
Load loaders

### LINKED ISSUE #2283: load_loaders is called inside execute transaction loop on every transaction
#### Problem
load_loaders is called inside execute transaction loop.  This causes the loop to take a lock to the account database on every transaction it executes.

#### Proposed Solution

take a lock once for the entire batch of transactions and load all the loaders.

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

#2283

#### Summary of Changes

Uses a single lock for account loading and consolidates the account loading error handling.

Fixes #2283

### CODE REVIEW

File: src/accounts.rs [hunk 1]
```diff
@@ -175,18 +178,67 @@ impl AccountsDB {
...
-        results: Vec<Result<()>>,
+        lock_results: Vec<Result<()>>,
         max_age: usize,
         error_counters: &mut ErrorCounters,
-    ) -> Vec<(Result<Vec<Account>>)> {
+    ) -> Vec<Result<(InstructionAccounts, InstructionLoaders)>> {
         txs.iter()
-            .zip(results.into_iter())
+            .zip(lock_results.into_iter())
             .map(|etx| match etx {
```
Comment (garious) [inline src/accounts.rs]: Can you either break this out into a separate function or find a way to make use of the `?` operator to get rid of all the pattern matching?

**Comment by aeyakovenko** [comment]
@jackcmay checkout how I thought load_loaders could work https://github.com/aeyakovenko/solana/blob/forking/src/accounts.rs

**Comment by jackcmay** [comment]
@aeyakovenko Looks like our changes are mostly compatible and complimentatry

File: src/accounts.rs (hunk 1 above)
Comment (jackcmay) [reply src/accounts.rs]: @garious, result matching gotten rid of :-)

**Comment by aeyakovenko** [comment]
@jackcmay great!  Feel free to yank that. I am probably not going to merge that whole branch. I am just trying to figure out what needs to break for checkpoints.

**Comment by jackcmay** [comment]
Mind if I land this pr?

**Pull request merged**